src/core_codemods/sql_parameterization.py

```diff
@@ -39,10 +56,57 @@
 raw_quote_pattern = re.compile(r"(?<!\\)'")
 
 
-class SQLQueryParameterizationTransformer(LibcstResultTransformer, UtilsMixin):
-    change_description = "Parameterized SQL query execution."
+class ExtractPrefixMixin(cst.MetadataDependent):
+
+    METADATA_DEPENDENCIES: ClassVar[Collection[ProviderT]] = (ParentNodeProvider,)
+
+    def extract_prefix(self, node: StringLiteralNodeType) -> str:
+        match node:
+            case cst.SimpleString():
+                return node.prefix.lower()
+            case cst.FormattedStringText():
+                try:
+                    parent = self.get_metadata(ParentNodeProvider, node)
+                    parent = cst.ensure_type(parent, cst.FormattedString)
+                except Exception:
+                    return ""
+                return parent.start.lower()
+            case PrintfStringText():
+                return self.extract_prefix(node.origin)
+        return ""
+
+    def _extract_prefix_raw_value(self, node: StringLiteralNodeType) -> tuple[str, str]:
+        raw_value = extract_raw_value(node)
+        if (prefix := self.extract_prefix(node)) is not None:
+            return prefix, raw_value
+        return prefix, raw_value
```

Comment:
maybe I'm missing something super subtle but can't you just
```suggestion
prefix = self.extract_prefix(node))
return prefix, raw_value
```
no if check, no double return